test/run_test.py

```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/env python
+
+import argparse
+import os
+import subprocess
+import sys
+
+TESTS = [
+    'autograd',
+    'cpp_extensions',
+    'cuda',
+    'dataloader',
+    'distributed',
+    'distributions',
+    'indexing',
+    'jit',
+    'legacy_nn',
+    'multiprocessing',
+    'nccl',
+    'nn',
+    'optim',
+    'sparse',
+    'torch',
+    'utils',
+]
+
+WINDOWS_BLACKLIST = [
+    'cpp_extensions',
+    'distributed',
+]
+
+
+def shell(command, cwd):
+    popen = subprocess.Popen(
+        command,
+        stdout=subprocess.PIPE,
+        universal_newlines=True,
+        cwd=cwd,
+        shell=True)
+    for stdout_line in iter(popen.stdout.readline, ''):
+        print(stdout_line.strip())
+    popen.stdout.close()
+    return_code = popen.wait()
+    return return_code == 0
+
+
+def get_shell_output(command):
+    return subprocess.check_output(command, shell=True).decode().strip()
+
+
+def run_test(python, test_module, cwd, verbose):
+    verbose = '--verbose' if verbose else ''
+    shell('{} {} {}'.format(python, test_module, verbose), cwd)
+
+
+def test_cpp_extensions(python, test_module, test_directory, verbose):
+    shell('{} setup.py install --root ./install'.format(python),
+          os.path.join(test_directory, 'cpp_extensions'))
+
+    python_path = os.environ.get('PYTHONPATH', '')
+    install_directory = get_shell_output(
+        "find cpp_extensions/install -name '*-packages'")
+    install_directory = os.path.join(test_directory, install_directory)
+    os.environ['PYTHONPATH'] = '{}:{}'.format(install_directory, python_path)
+    run_test(python, test_module, test_directory, verbose)
+    os.environ['PYTHONPATH'] = python_path
+
+
+def test_distributed(python, test_module, cwd, verbose):
+    os.environ['PYCMD'] = python
+    shell('./run_distributed_tests.sh', cwd)
```

Comment:
is `cwd` the same as `test_directory` above (and as `cwd` in `run_test`)? It's better to use the same name everywhere

Reply:
Yes thanks